.github/workflows/build-test.yml

```diff
@@ -0,0 +1,37 @@
+name: Go-build-status
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.15
```

Comment:
Would we want to use a newer version of golang?

Reply:
Upgraded to 1.17